### PR TITLE
FS-387 hello world security

### PR DESCRIPTION
**Pull request by ramsharma-prog**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+@pytest.fixture()
+def flask_test_client():
+    """
+    Creates the test client we will be using to test the responses
+    from our app, this is a test fixture.
+    :return: A flask test client.
+    """
+    with app.test_client() as test_client:
+        yield test_client
```
Comment (SentientHousePlant) [inline tests/conftest.py:13]: Are your precommit hooks installed? This file isn't pep8 compliant, which is a nitpick but can lead to inconsistent styling over time within the dev team.

File: tests/test_security_files.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+def test_insecure_post_fails(flask_test_client):
+  response = flask_test_client.post(
+    '/',
+    data = dict(email="test@company.com", password="test"),
```
Comment (SentientHousePlant) [inline tests/test_security_files.py:4]: Silly question but why is any data posted?

File: tests/test_security_files.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+def test_insecure_post_fails(flask_test_client):
```
Comment (SentientHousePlant) [inline tests/test_security_files.py:1]: I am not very sharp - do you think you could explain this test a little bit more? Thank you.

**Review by SentientHousePlant — COMMENTED** [review]
Do add some explanations/comments for the unexperienced of us on the team :) thank you.

File: tests/test_security_files.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+  response = flask_test_client.post(
+    '/',
+    data = dict(email="test@company.com", password="test"),
+    follow_redirects=True
+    )
+  assert response.status_code == 403
```
Comment (Adam-W1) [inline tests/test_security_files.py:7]: Do we need to test both HSTS security (enforced TCP with the introduction of strict_transport_security) and CSRF preventing cross-site request forgery (with the introduction of SeaSurf) here?

File: tests/conftest.py (hunk 1 above)
Comment (Adam-W1) [reply tests/conftest.py:13]: if helpful theres also a plugin for on the go formatting, i now have the following set up in VScode user settings:
    "python.formatting.blackArgs": ["--line-length", "120"], // 120 may not be in line with pre commit config
    "python.formatting.provider": "black"

File: tests/test_security_files.py (hunk 4 above)
Comment (SentientHousePlant) [reply tests/test_security_files.py:7]: Could HSTS be tested by making a http (not https) request checking it fails? We are currently (very implicitly and manually) testing this when we attempt to use flask in production mode - since our local dev server is not https- because it fails all requests.

File: tests/conftest.py (hunk 1 above)
Comment (SentientHousePlant) [reply tests/conftest.py:13]: One can also add black to PyCharm - please see the following [link](https://black.readthedocs.io/en/stable/integrations/editors.html) . As expected, VSCode is more elegant than Pycharm.

File: tests/test_security_files.py (hunk 4 above)
Comment (Adam-W1) [reply tests/test_security_files.py:7]: yes i think the http test expecting failure would cover HSTS - always good to have the explicit test

File: app/create_app.py [hunk 5]
```diff
@@ -2,8 +2,12 @@
 from flask_compress import Compress
 from flask_talisman import Talisman
 from jinja2 import ChoiceLoader, PackageLoader, PrefixLoader
+from flask_seasurf import SeaSurf
 
+csrf = SeaSurf()
 app = Flask(__name__, static_url_path="/assets")
+app.config['SECRET_KEY'] = '087bc8e3108ceae8f7864d76cbf6408a'
```
[the remark this answers is not in the document]
Comment (ramsharma-prog) [reply app/create_app.py:9]: Created two separate classes for development and production

File: tests/test_security_files.py (hunk 3 above)
Comment (stevewalton28) [inline tests/test_security_files.py:1]: File name not correct - test_security_files isn't really what its doing.  Need a better more descriptive name than that "files" is the offending word I think

**Review by stevewalton28 — CHANGES_REQUESTED** [review]

File: tests/test_csrf.py [hunk 6]
```diff
@@ -0,0 +1,7 @@
+def test_csrf_files(flask_test_client):
```
Comment (stevewalton28) [inline tests/test_csrf.py:1]: This test name doesn't make sense to me - what does the "files" bit represent?

File: tests/test_csrf.py [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+    """
+    The test should return unauthorised message
+     if the request is not received from https server
```
Comment (stevewalton28) [inline tests/test_csrf.py:4]: Is this true?  I'm not sure this test belongs with other CSRF tests.  What is the purpose of this test?

File: requirements.txt [hunk 8]
```diff
@@ -3,7 +3,7 @@ Brotli==1.0.9
 cfgv==3.3.1
 click==8.0.3
 distlib==0.3.4
-filelock==3.4.2
+filelock>=3.4.1
```
Comment (stevewalton28) [inline requirements.txt:6]: Why the downgrade?

File: manifest.yml [hunk 9]
```diff
@@ -0,0 +1,7 @@
+---
+applications:
+- name: funding-service-design-frontend
+  memory: 64M
+  buildpacks:
+  - https://github.com/cloudfoundry/python-buildpack.git
+  command: flask run --host 0.0.0.0 --port 8080
```
Comment (stevewalton28) [inline manifest.yml:7]: New line required at end of file

**Review by stevewalton28 — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 8 above)
Comment (ramsharma-prog) [reply requirements.txt:6]: It looks like the file got downgraded  during the branch merge. I will fix it

File: tests/test_csrf.py (hunk 7 above)
Comment (ramsharma-prog) [reply tests/test_csrf.py:4]: Sure, I will correct it

File: tests/test_csrf.py (hunk 6 above)
Comment (ramsharma-prog) [reply tests/test_csrf.py:1]: I will change the name to more specific

File: tests/test_csrf.py (hunk 7 above)
Comment (stevewalton28) [reply tests/test_csrf.py:4]: But what is the purpose of the test Ram?

File: app/create_app.py [hunk 10]
```diff
@@ -1,14 +1,26 @@
+import os
+
 from flask import Flask
 from flask_compress import Compress
+from flask_seasurf import SeaSurf
 from flask_talisman import Talisman
-from jinja2 import ChoiceLoader, PackageLoader, PrefixLoader
+from jinja2 import ChoiceLoader
```
Comment (endamccormack) [inline app/create_app.py:7]: why arent these on 1 line?

File: app/create_app.py [hunk 11]
```diff
@@ -1,14 +1,26 @@
...
+from jinja2 import PackageLoader
+from jinja2 import PrefixLoader
+
 
 app = Flask(__name__, static_url_path="/assets")
+app.config["SECRET_KEY"] = os.environ.get("CSRF_KEY")
```
Comment (endamccormack) [inline app/create_app.py:13]: why is this being set here and not in the config file?

File: tests/test_csrf.py (hunk 7 above)
Comment (ramsharma-prog) [reply tests/test_csrf.py:4]: I was trying to write a test for unauthorised entry

**Review by endamccormack — CHANGES_REQUESTED** [review]

File: app/create_app.py (hunk 10 above)
Comment (ramsharma-prog) [reply app/create_app.py:7]: Added on the same line

File: app/create_app.py (hunk 11 above)
Comment (ramsharma-prog) [reply app/create_app.py:13]: Moved it to config.py file

File: tests/test_csrf.py (hunk 7 above)
Comment (ramsharma-prog) [reply tests/test_csrf.py:4]: Got rid of this test file for this sprint.  I will try to write a test on later sprints. OWASP zap scan added to .github

File: tests/test_csrf.py (hunk 6 above)
Comment (ramsharma-prog) [reply tests/test_csrf.py:1]: Got rid of this test file for this sprint. I will try to write a test on later sprints. OWASP zap scan added to .github

**Comment by endamccormack** [comment]
Please create a story in jira to ensure that stray test is removed later

File: .github/workflows/govcloud.yml [hunk 12]
```diff
@@ -0,0 +1,79 @@
+name: Deploy to Cloud Foundry
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+    - master
+    - FS-389
```
[the remark this answers is not in the document]
Comment (SentientHousePlant) [reply .github/workflows/govcloud.yml]: Should be:
python -m pip install --upgrade pip && python -m pip install -r requirements.txt
under security.

File: .github/workflows/govcloud.yml [hunk 13]
```diff
@@ -35,9 +35,8 @@ jobs:
         with:
         # Optional: do not specify to match Chrome's version
           chromedriver-version: '88.0.4324.96'
-      # - name: Run Accessability tests
-      #   run: pytest --driver Chrome --driver-path .venv/lib/python3.10/site-packages/chromedriver_py/chromedriver_linux64
-
+#      - name: Run Accessability tests
```
Comment (endamccormack) [inline .github/workflows/govcloud.yml:38]: why is this here

**Review by endamccormack — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 14]
```diff
@@ -131,3 +131,8 @@ dmypy.json
 .idea
 app/static
 govuk_components
+
+# bash_profile & .DS_Store
```
Comment (endamccormack) [inline .gitignore:135]: comment

**Review by endamccormack — CHANGES_REQUESTED** [review]

File: .github/workflows/govcloud.yml (hunk 13 above)
Comment (ramsharma-prog) [reply .github/workflows/govcloud.yml:38]: I commented this out as my branch doesn't have harry's files so the accessibility test was failing during checks

File: .gitignore (hunk 14 above)
Comment (ramsharma-prog) [reply .gitignore:135]: Hello world FS-387 would be fine?

Comment (ramsharma-prog) [reply .gitignore:135]: Hello world security?

File: .github/workflows/govcloud.yml (hunk 13 above)
Comment (ramsharma-prog) [reply .github/workflows/govcloud.yml]: Resolved all conflicts  & uncommented above lines

File: .gitignore (hunk 14 above)
Comment (ramsharma-prog) [reply .gitignore]: Commented: # Hello world security FS-387 files

**Review by endamccormack — APPROVED** [review]

**Review by stevewalton28 — APPROVED** [review]
